### ISSUE TITLE
Question Regarding Benchmark

### ISSUE THREAD

**Issue by yechunjin** [Hoar012/RAP-MLLM#3]
Thank you for your great work! 

The results are testing against yollava's benchmark, but yollava seems to provide only 171 VQA questions and not the remaining 400 text-only QA questions. Am I missing something?

**Comment by Hoar012** [comment]
Thanks for your interest in our work! You can find the data at this [Google Drive link](https://drive.google.com/drive/folders/1bqM5y0-Kw26R5T4kfaeUAZzeKqIdREdU?usp=sharing).

**Comment by yechunjin** [comment]
Thank you for the prompt response!

**Comment by yechunjin** [comment]
> Thanks for your interest in our work! You can find the data at this [Google Drive link](https://drive.google.com/drive/folders/1bqM5y0-Kw26R5T4kfaeUAZzeKqIdREdU?usp=sharing).

I am trying to create the complete database using this [Google Drive link](https://drive.google.com/drive/folders/1bqM5y0-Kw26R5T4kfaeUAZzeKqIdREdU?usp=sharing) for all objects of yollava and then test the VQA and TQA task. 

The accuracy of LLaVA+Retriever was 0.917 (0.912 reported in the paper) in the VQA task, and 0.847 (0.863 reported in the paper) in the TQA task.
What puzzles me, however, is that the accuracy of RAP-MLLM was 0.747 (0.935 reported in the paper) in the VQA task, and 0.932 (0.938 reported in the paper) in the TQA task.

Why does the RAP-MLLM degrade so much in performance in the VQA task?

Am I missing something?

python eval/VQA.py --eval-file /home/ycj/YoLLaVA/yollava-data/yollava-visual-qa.json  --model-path Hoar012/RAP-LLaVA-13b  --retrieval --database "/home/ycj/RAP-MLLM/example_database/" --topK 1

**Comment by Hoar012** [comment]
Thanks for your feedback. I have tried to reproduce the problem, but I did not encounter the issue you mentioned. Could you provide more details about your implementation, such as the approach used for database construction?

**Comment by yechunjin** [comment]
> Thanks for your feedback. I have tried to reproduce the problem, but I did not encounter the issue you mentioned. Could you provide more details about your implementation, such as the approach used for database construction?

For the database construction, I used subject-detailed-captions.json (from this [Google Drive link](https://drive.google.com/drive/folders/1bqM5y0-Kw26R5T4kfaeUAZzeKqIdREdU?usp=sharing)) to build the description and selected yollava-data/train/<sks>/0.png as the reference image for each concept. Here's the database I built. [database link](https://drive.google.com/file/d/1IBs_itP59bI9a_6WbMPjqHz_TLSgFH19/view?usp=drive_link)

If I understand correctly, the difference between LLaVA+Retriever and RAP-MLLM is whether the model accepts personalized fine-tuning. But, in the VQA task, the accuracy of LLaVA+Retriever is normal. This confuses me.

**Comment by Hoar012** [comment]
> > Thanks for your feedback. I have tried to reproduce the problem, but I did not encounter the issue you mentioned. Could you provide more details about your implementation, such as the approach used for database construction?
> 
> For the database construction, I used subject-detailed-captions.json (from this [Google Drive link](https://drive.google.com/drive/folders/1bqM5y0-Kw26R5T4kfaeUAZzeKqIdREdU?usp=sharing)) to build the description and selected yollava-data/train/sks/0.png as the reference image for each concept. Here's the database I built. [database link](https://drive.google.com/file/d/1IBs_itP59bI9a_6WbMPjqHz_TLSgFH19/view?usp=drive_link)
> 
> If I understand correctly, the difference between LLaVA+Retriever and RAP-MLLM is whether the model accepts personalized fine-tuning. But, in the VQA task, the accuracy of LLaVA+Retriever is normal. This confuses me.

I have carefully checked the database you provided and used it for evaluation, but I still did not encounter the issue. I obtained an accuracy of 0.9 on the VQA task.

The only potential issue I found is that the subject-detailed captions are relatively long as text information, which may slightly impact the model's performance. For easier reproduction of our results, I can share the [database](https://drive.google.com/drive/folders/1VI6cxj10Owp6uTCqNAYPJj86zCnMMR7X?usp=sharing) used in our experiment.

Perhaps you could check the model’s output in the result file to help pinpoint the problem.

**Comment by yechunjin** [comment]
> > > Thanks for your feedback. I have tried to reproduce the problem, but I did not encounter the issue you mentioned. Could you provide more details about your implementation, such as the approach used for database construction?
> > 
> > 
> > For the database construction, I used subject-detailed-captions.json (from this [Google Drive link](https://drive.google.com/drive/folders/1bqM5y0-Kw26R5T4kfaeUAZzeKqIdREdU?usp=sharing)) to build the description and selected yollava-data/train/sks/0.png as the reference image for each concept. Here's the database I built. [database link](https://drive.google.com/file/d/1IBs_itP59bI9a_6WbMPjqHz_TLSgFH19/view?usp=drive_link)
> > If I understand correctly, the difference between LLaVA+Retriever and RAP-MLLM is whether the model accepts personalized fine-tuning. But, in the VQA task, the accuracy of LLaVA+Retriever is normal. This confuses me.
> 
> I have carefully checked the database you provided and used it for evaluation, but I still did not encounter the issue. I obtained an accuracy of 0.9 on the VQA task.
> 
> The only potential issue I found is that the subject-detailed captions are relatively long as text information, which may slightly impact the model's performance. For easier reproduction of our results, I can share the [database](https://drive.google.com/drive/folders/1VI6cxj10Owp6uTCqNAYPJj86zCnMMR7X?usp=sharing) used in our experiment.
> 
> Perhaps you could check the model’s output in the result file to help pinpoint the problem.

Thanks a lot.
